### PR TITLE
s2s VPN between two VNets

### PR DESCRIPTION
**Pull request by fabferri**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [X ] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog
ARM template creates two Site-to-Site VPN tunnels between two Azure Virtual Networks. In each Azure VNet is deployed an Azure VPN Gateway in configuration active-active in availability zones. The Azure VPN Gateway advertises through BGP the Azure Virtual network address space to the remote peer.
*
*
*

### CODE REVIEW

**Comment by fabferri** [comment]
Hello, 
please let me know if all is OK to merge the pull request.
Thanks

File: 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,1109 @@
...
+      }
+    },
+    "dnsLabelgtw2PubIP2": {
+      "type": "string",
+      "defaultValue": "[toLower( concat( 'gtw2-ip2-', uniqueString(resourceGroup().id) ))]",
+      "metadata": {
+        "description": "dns name of public IP2 of the VPN Gateway2. Must be lowercase. It should match with the regex: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$."
+      }
+    },
+    "imagePublisher": {
```
Comment (bmoore-msft) [inline 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json:374]: set some allowed values on the publisher, offer, sku params

File: 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,1109 @@
...
+      "metadata": {
+        "description": "administrator password of the VMs"
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": [
+        "Standard_LRS",
+        "Premium_LRS"
```
Comment (bmoore-msft) [inline 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json:340]: if you're going to allow for premium disk then the VM sizes must match... and alternative is to use StandardSSD which is supported by all VM sizes

File: 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,1109 @@
...
+    "storageAccountType": "[parameters('storageAccountType')]"
+  },
+  "resources": [
+    {
+
+      "type": "Microsoft.Network/networkSecurityGroups",
+      "name": "[variables('nsg1')]",
+      "apiVersion": "2020-08-01",
+      "location": "[variables('location1')]",
+      "comments": "network security group applied to the subnets in the VNet1",
```
Comment (bmoore-msft) [inline 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json:463]: See: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#sort-order-of-properties for the proper sort order of properties

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json (hunk 3 above)
Comment (fabferri) [reply 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json]: I have changed the proper sort order as in best practice

File: 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json (hunk 1 above)
Comment (fabferri) [reply 201-site-to-site-vpn-fqdn-bgp/azuredeploy.json]: added the variable:
"imageReference": {
"publisher": "Canonical",
"offer": "0001-com-ubuntu-server-focal",
"sku": "20_04-lts",
"version": "latest"
},

**Comment by fabferri** [comment]
changes applied. please review

**Pull request merged**